src/uct/ib/base/ib_device.c

```diff
@@ -690,21 +700,25 @@ int uct_ib_device_is_port_roce(uct_ib_device_t *dev, uint8_t port_num)
     return IBV_PORT_IS_LINK_LAYER_ETHERNET(uct_ib_device_port_attr(dev, port_num));
 }
 
-ucs_status_t uct_ib_device_select_gid_index(uct_ib_device_t *dev,
-                                            uint8_t port_num,
-                                            size_t md_config_index,
-                                            uint8_t *gid_index)
+ucs_status_t uct_ib_device_select_roce_gid_index(uct_ib_device_t *dev,
```

Comment:
rename to uct_ib_iface_init_roce_gid_info(iface, config), move to ib_iface.c, it will allow reducing some code

Reply:
will need to create a dummy iface or something in gtest then - https://github.com/openucx/ucx/blob/master/test/gtest/uct/ib/test_ib.cc#L300

Reply:
@yosefe in the gtest, it needs to get a gid_index from a given port (m_port) so i need to have an iface open on this specific port. so creating a dummy iface won't help since i cant control the port it will open on.

Reply:
can you create a dummy uct_iface_t and set any needed fields in it manually, such as port number